packages/next/next-server/lib/router/utils/parse-relative-url.ts

```diff
@@ -0,0 +1,24 @@
+const DUMMY_BASE = new URL('http://n')
+
+/**
+ * Parses path-relative urls (e.g. `/hello/world?foo=bar`). If url isn't path-relative
+ * (e.g. `./hello`) then at least base must be.
+ * Absolute urls are rejected.
+ */
+export function parseRelativeUrl(url: string, base?: string) {
+  const resolvedBase = base ? new URL(base, DUMMY_BASE) : DUMMY_BASE
+  const { pathname, searchParams, search, hash, href, origin } = new URL(
+    url,
+    resolvedBase
+  )
+  if (origin !== DUMMY_BASE.origin) {
```

Comment:
@Janpot This clause triggers when I'm using the `assetPrefix` option in production. Is this expected?

Reply:
are you using an absolute assetPrefix?

Reply:
It's absolute, yes: 

```      assetPrefix: isProd ? "https://d34vhhr1t6sulw.cloudfront.net" : "",```

No minimal production but you can see the error here: https://d34vhhr1t6sulw.cloudfront.net/archiv

Reply:
@Janpot Could you take a look at it yet?

Reply:
@stephanschubert We are following up at https://github.com/vercel/next.js/issues/15188